examples/esp32/xively_demo/main/xively_task.h

```diff
@@ -3,90 +3,167 @@
  * This is part of the Xively C Client codebase,
  * it is licensed under the BSD 3-Clause license.
  */
-#ifndef __XIVELY_IF_H__
-#define __XIVELY_IF_H__
+#ifndef __XIVELY_TASK_H__
+#define __XIVELY_TASK_H__
+
+#include <stdint.h>
+#include "xively.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
+/*! \file
+ * @brief Interface between the top level application logic and libxively for
+ * an RTOS-based environment.
+ *
+ * \mainpage Sample Implementation of a Dedicated RTOS Task for libxively
+ *
+ * @detailed Xively Task is a state machine that will run on its own RTOS task,
+ * and a task-safe API to communicate with it, request the task to pause/shutdown
+ * when other parts of the application need extra resources, etc.
+ *
+ * Most, if not all of the libxively logic is run from within this task, so this
+ * demo can be used to measure the stack requirements of the library.
+ *
+ * For some production systems, you may want/need to implement a limit on how many
+ * subscribe/publish messages are in flight at any given point; not sending the
+ * next publish/subscribe message until the callback for the previous one(s)
+ * have acknowledged success.
+ * This will help make sure libxively doesn't internally allocate too much memory
```

Comment:
maybe for next iteration, but w're trying to consistently call it Xively Client Library instead of libxively.